samples/net/sockets/socketpair/src/socketpair_example.c

```diff
@@ -110,8 +111,10 @@ int main(int argc, char *argv[])
 
 	for (i = 0; i < ARRAY_SIZE(ctx); ++i) {
 		ctx[i].name = (char *)names[i];
-		socketpair(AF_UNIX, SOCK_STREAM, 0, ctx[i].spair);
-		pthread_create(&ctx[i].thread, NULL, fun, &ctx[i]);
+		r = socketpair(AF_UNIX, SOCK_STREAM, 0, ctx[i].spair);
+		assert(r == 0);
```

Comment:
This should probably be `__ASSERT_NO_MSG()`, which is the Zephyr way of doing this.

Reply:
Yeah, that won't work for the Makefile.posix way of building though.

Reply:
Why not?  There are uses of Zephyr `__ASSERT()` in lib/posix.

Reply:
@pabigot, Makefile.posix is for running this sample code entirely without Zephyr as a native Linux (or BSD, or macOS) application. I.e. lib/posix is nonexistent.

Reply:
A possible alternative solution that doesn't use any asserts is to just use a goto to go to the end of the function. There is no resource cleanup, but again, it's just a sample application to illustrate usage.

Reply:
Or, if you really don't care, use `(void)socketpair(...)`.

Reply:
I should probably care. Just remembered that `pthread_create()` requires a `pthread_attr_t` in Zephyr, so that needs to be modified anyway.

Reply:
I think assert() is fine here, and @cfriedt, thanks for providing Makefile.posix - that's the way we can ensure that we actually implement POSIX API portable across OSes. (Maybe one sweet day we'll be able to do the same for tests, but examples with Makefile.posix are the baseline to stick to.)

Reply:
@cfriedt: Also, FYI (not just for this code piece, but in general), there's an option of marking Coverity case as false positive. I myself don't remember practicing it much, but see e.g. @jukkar practicing it regularly (no implications meant). So, if somewhere your code is correct, but Coverity doesn't get it, you can comment on the initial ticket that it's a false positive (someone would need to propagate that back to Coverity UI).